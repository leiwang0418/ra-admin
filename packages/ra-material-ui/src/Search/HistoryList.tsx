// import { useState } from "react";
// import PropTypes, { InferProps } from "prop-types";
// import { defineMessages, useIntl } from "react-intl";
// import { styled } from "@material-ui/core/styles";
// import MenuList from "@material-ui/core/MenuList";
// import ListItemText from "@material-ui/core/ListItemText";
// import Typography from "@material-ui/core/Typography";
// import ListItemIcon from "@material-ui/core/ListItemIcon";
// import CloseIcon from "@material-ui/icons/Close";
// import HistoryIcon from "@material-ui/icons/History";
// import SearchItem from "./SearchItem";
// import LightTooltip from "../LightTooltip";

// const RemoveItemIcon = styled(ListItemIcon)(() => ({
// 	cursor: "pointer",
// }));

// const HistoryListPropTypes = {
// 	items: PropTypes.arrayOf(
// 		PropTypes.shape({
// 			id: PropTypes.number.isRequired,
// 			content: PropTypes.string.isRequired,
// 		})
// 	),
// };

// export type HistoryListTypes = InferProps<typeof HistoryListPropTypes>;

// const HistoryList = (props: HistoryListTypes) => {
// 	// const { items } = props;
// 	const removeLabel = messages.removeLabel;
// 	const intl = useIntl();
// 	const [mouseEnter, setMouseEnter] = useState<boolean>(false);

// 	return (
// 		<>
// 			<SearchItem
// 				onMouseEnter={() => setMouseEnter(true)}
// 				onMouseLeave={() => setMouseEnter(false)}
// 			>
// 				<ListItemIcon>
// 					<HistoryIcon fontSize="small" />
// 				</ListItemIcon>
// 				<ListItemText>
// 					<Typography variant="inherit" noWrap>
// 						A very long text that overflows Add space after
// 						paragraphAdd space after paragraphAdd space after
// 						paragraphAdd space after paragraphAdd space after
// 						paragraphAdd space after paragraph
// 					</Typography>
// 				</ListItemText>
// 				<LightTooltip
// 					followCursor
// 					placement="right"
// 					title={removeLabel ? intl.formatMessage(removeLabel) : ""}
// 				>
// 					<RemoveItemIcon
// 						sx={{ visibility: mouseEnter ? "visible" : "hidden" }}
// 						onClick={() => console.log("remove history")}
// 					>
// 						<CloseIcon fontSize="small" />
// 					</RemoveItemIcon>
// 				</LightTooltip>
// 			</SearchItem>
// 		</>
// 	);
// };

// HistoryList.propTypes = HistoryListPropTypes;

// const messages = defineMessages({
// 	removeLabel: {
// 		id: "ra.defaultSearch.remove.label",
// 		defaultMessage: "Remove from history",
// 	},
// });

// export default HistoryList;
