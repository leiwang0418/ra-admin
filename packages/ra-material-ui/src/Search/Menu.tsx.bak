import PropTypes, { InferProps } from "prop-types";
import { defineMessages, useIntl } from "react-intl";
import { styled } from "@material-ui/core/styles";
import MenuList from "@material-ui/core/MenuList";
import MenuItem from "@material-ui/core/MenuItem";
import ListItemText from "@material-ui/core/ListItemText";
import Typography from "@material-ui/core/Typography";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import CloseIcon from "@material-ui/icons/Close";
import Tooltip, {
	TooltipProps,
	tooltipClasses,
} from "@material-ui/core/Tooltip";
import HistoryIcon from "@material-ui/icons/History";

const RemoveItemIcon = styled(ListItemIcon)(() => ({
	cursor: "pointer",
}));

const LightTooltip = styled(({ className, ...props }: TooltipProps) => (
	<Tooltip {...props} classes={{ popper: className }} />
))(({ theme }) => ({
	[`& .${tooltipClasses.tooltip}`]: {
		backgroundColor: theme.palette.common.white,
		color: "rgba(0, 0, 0, 0.87)",
		boxShadow: theme.shadows[1],
		fontSize: 11,
	},
}));

const MenuPropTypes = {
	isOpen: PropTypes.bool,
	removeLabel: PropTypes.object,
	items: PropTypes.shape({
		config: PropTypes.shape({
			icon: PropTypes.element
			,
		}),
		data: PropTypes.arrayOf(
			PropTypes.shape({
				id: PropTypes.number.isRequired,
				content: PropTypes.string.isRequired,
			})
		),
	}),
};

export type MenuTypes = InferProps<typeof MenuPropTypes>;

const Menu = (props: MenuTypes) => {
	const {
		isOpen = true,
		removeLabel = messages["ra.defaultSearch.remove.label"],
	} = props;
	const intl = useIntl();

	if (!isOpen) {
		return null;
	}
	return (
		<MenuList>
			<SearchItem>
				<ListItemIcon>
					<HistoryIcon fontSize="small" />
				</ListItemIcon>
				<ListItemText>
					<Typography variant="inherit" noWrap>
						A very long text that overflows Add space after
						paragraphAdd space after paragraphAdd space after
						paragraphAdd space after paragraphAdd space after
						paragraphAdd space after paragraph
					</Typography>
				</ListItemText>
				<LightTooltip
					followCursor
					placement="right"
					title={removeLabel ? intl.formatMessage(removeLabel) : ""}
				>
					<RemoveItemIcon
						onClick={() => console.log("remove history")}
					>
						<CloseIcon fontSize="small" />
					</RemoveItemIcon>
				</LightTooltip>
			</SearchItem>
			<SearchItem>Custom spacing</SearchItem>
		</MenuList>
	);
};

Menu.propTypes = MenuPropTypes;

const messages = defineMessages({
	"ra.defaultSearch.remove.label": {
		id: "ra.defaultSearch.remove.label",
		defaultMessage: "Remove from history",
	},
});

export default Menu;
