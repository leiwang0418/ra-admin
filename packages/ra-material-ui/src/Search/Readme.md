basic Search

```js
const top100Films = [
  "The Godfather: Part II",
  "One Flew Over the Cuckoo's Nest",
  "Goodfellas",
  "The Matrix",
  "Seven Samurai",
  "Star Wars: Episode IV - A New Hope",
];

<Search id="testSearch" width={500} options={top100Films} />;
```
