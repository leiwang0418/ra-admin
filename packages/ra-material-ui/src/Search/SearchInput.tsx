// import PropTypes, { InferProps } from "prop-types";
// import { defineMessages, useIntl } from "react-intl";
// import {
// 	alpha,
// 	Divider,
// 	IconButton,
// 	InputAdornment,
// 	InputBase,
// 	Paper,
// 	PaperProps,
// } from "@mui/material";
// import SearchIcon from "@mui/icons-material/Search";
// import CancelRoundedIcon from "@mui/icons-material/CancelRounded";
// import TuneIcon from "@mui/icons-material/Tune";
// import { styled } from "@mui/system";
// import { Theme } from "@mui/material";

// interface StyledPaperProps extends PaperProps {
// 	focused?: boolean;
// 	theme?: Theme;
// }

// const SearchRoot = styled(Paper, {
// 	shouldForwardProp: (prop) => prop !== "focused",
// })<StyledPaperProps>(({ focused, theme }) => {
// 	return {
// 		borderRadius: theme.shape.borderRadius,
// 		padding: "2px 4px",
// 		alignItems: "center",
// 		backgroundColor: focused
// 			? theme.palette.common.white
// 			: alpha(theme.palette.common.black, 0.1),
// 		...(focused && { boxShadow: theme.shadows[2] }),
// 	};
// });

// const SearchEndAdornment = styled("div")({
// 	// We use a position absolute to support wrapping tags.
// 	position: "absolute",
// 	right: 0,
// 	display: "contents",
// });

// const SearchPropTypes = {
// 	placeholderLabel: PropTypes.object,
// 	clearLabel: PropTypes.object,
// 	searchLabel: PropTypes.object,
// 	showSearchOptions: PropTypes.object,
// 	// focus: PropTypes.bool.isRequired,
// 	params: PropTypes.object,
// };

// export type SearchTypes = InferProps<typeof SearchPropTypes>;

// const SearchInput = (props: SearchTypes) => {
// 	const {
// 		placeholderLabel = messages.placeholder,
// 		clearLabel = messages.clearLabel,
// 		searchLabel = messages.searchLabel,
// 		showSearchOptions = messages.showSearchOptions,
// 		focused,
// 		hasClearIcon,
// 		getClearProps,
// 		hasSearchOptions,
// 		getRootProps,
// 		getInputProps,
// 	} = props;

// 	console.log("getClearProps()", getClearProps());

// 	const intl = useIntl();

// 	const renderClearIcon = (
// 			<IconButton
// 				{...getClearProps()}
// 				aria-label={clearLabel ? intl.formatMessage(clearLabel) : ""}
// 			>
// 				<CancelRoundedIcon />
// 			</IconButton>
// 	);
// 	const renderDriver = (<Divider
// 				sx={{
// 					height: 20,
// 					m: 0.5,
// 				}}
// 				orientation="vertical"
// 			/>);


// 	const renderSearch = (
// 		<InputAdornment position="start">
// 			<IconButton
// 				aria-label={searchLabel ? intl.formatMessage(searchLabel) : ""}
// 				onClick={() => console.log("basic search")}
// 			>
// 				<SearchIcon onMouseDown={(e) => e.preventDefault()} />
// 			</IconButton>
// 		</InputAdornment>
// 	);
// 	const renderShowSearchOptions = (
// 		<IconButton
// 			onClick={() => console.log("ShowSearchOptions")}
// 			sx={{ p: "5px" }}
// 			aria-label={
// 				showSearchOptions ? intl.formatMessage(showSearchOptions) : ""
// 			}
// 		>
// 			<TuneIcon />
// 		</IconButton>
// 	);

// 	return (
// 		<SearchRoot
// 			{...getRootProps()}
// 			focused={focused}
// 			elevation={focused ? 1 : 0}
// 		>
// 			<InputBase
// 				fullWidth
// 				startAdornment={renderSearch}
// 				endAdornment={
// 					<SearchEndAdornment>
// 						{hasClearIcon ? renderClearIcon : null}
// 						{hasClearIcon ? renderDriver : null}
// 						{renderShowSearchOptions}
// 					</SearchEndAdornment>
// 				}
// 				inputProps={{
// 					"aria-label": placeholderLabel
// 						? intl.formatMessage(placeholderLabel)
// 						: "",
// 					placeholder: placeholderLabel
// 						? intl.formatMessage(placeholderLabel)
// 						: "",
// 					...getInputProps(),
// 				}}
// 			/>
// 		</SearchRoot>
// 	);
// };

// SearchInput.propTypes = SearchPropTypes;

// const messages = defineMessages({
// 	placeholder: {
// 		id: "ra.defaultSearch.placeholder",
// 		defaultMessage: "Search...",
// 	},
// 	searchLabel: {
// 		id: "ra.defaultSearch.search.label",
// 		defaultMessage: "Search",
// 	},
// 	clearLabel: {
// 		id: "ra.defaultSearch.clear.label",
// 		defaultMessage: "Clear search",
// 	},
// 	showSearchOptions: {
// 		id: "ra.defaultSearch.show.search.options.label",
// 		defaultMessage: "Show search options",
// 	},
// });

// export default SearchInput;
