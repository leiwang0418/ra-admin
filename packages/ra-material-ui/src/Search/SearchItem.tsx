import { styled } from "@material-ui/core/styles";
import MenuItem from "@material-ui/core/MenuItem";

const SearchItem = styled(MenuItem)(({ theme }) => ({
	borderLeft: "2px solid #fff",
	cursor: "default",
	"&:hover": {
		borderLeftColor: theme.palette.primary.light,
	},
}));

export default SearchItem;