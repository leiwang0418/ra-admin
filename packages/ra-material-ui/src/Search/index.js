export {default as Search} from './Search';
// export {default as SearchInput} from './SearchInput';
