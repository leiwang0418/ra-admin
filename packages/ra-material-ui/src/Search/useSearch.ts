import React, { SyntheticEvent } from "react";
import {
	unstable_setRef as setRef,
	unstable_useEventCallback as useEventCallback,
	unstable_useControlled as useControlled,
	unstable_useId as useId,
} from "@mui/utils";

interface useSearchProps {
	componentName?: string;
	id: string;
	open?: boolean;
}

// 参考：https://github.com/mui-org/material-ui/blob/master/packages/mui-core/src/AutocompleteUnstyled/useAutocomplete.js
const useSearch = (props: useSearchProps) => {
	const { componentName = "useSearch", id: idProp, open: openProp } = props;

	const [open, setOpenState] = useControlled({
		controlled: openProp,
		default: false,
		name: componentName,
		state: "open",
	});

	const listboxAvailable = open; // 过滤列表长度大于0
	const popupOpen = open;
	const id = useId(idProp);
	const [anchorEl, setAnchorEl] = React.useState(null);

	return {
		getRootProps: (other = {}) => ({
			"aria-owns": listboxAvailable ? `${id}-listbox` : "",
			role: "combobox",
			"aria-expanded": listboxAvailable,
			onKeyDown: handleKeyDown(other),
			// onMouseDown: handleMouseDown,
			// onClick: handleClick,
		}),
		getInputProps: () => ({}),
		getClearProps: () => ({}),
		getListboxProps: () => ({}),
		focused: false,
		dirty: false,
		setAnchorEl,
	};
};

export default useSearch;
