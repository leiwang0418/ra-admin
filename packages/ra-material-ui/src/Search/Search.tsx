import React from "react";
import { useAutocomplete } from "@mui/core/AutocompleteUnstyled";
import PropTypes, { InferProps } from "prop-types";
import { defineMessages, useIntl } from "react-intl";
import {
  alpha,
  Divider,
  IconButton,
  InputAdornment,
  InputBase,
  ListItemIcon,
  MenuItem,
  Paper,
  PaperProps,
  ListItemText,
  Typography,
  Tooltip,
  TooltipProps,
  tooltipClasses,
  Popper,
} from "@mui/material";
import {
  Search as SearchIcon,
  CancelRounded as CancelRoundedIcon,
  Tune as TuneIcon,
  History as HistoryIcon,
  Close as CloseIcon,
} from "@mui/icons-material";
import { styled } from "@mui/system";
import { Theme } from "@mui/material";
import useSearch from "../SearchUnstyled/userSearch";
import setDefaults from "../utils/setDefaults";

interface StyledPaperProps extends PaperProps {
  focused?: boolean;
  theme?: Theme;
}

const SearchRoot = styled(Paper, {
  shouldForwardProp: (prop) => prop !== "focused",
})<StyledPaperProps>(({ focused, theme }) => {
  return {
    borderRadius: theme.shape.borderRadius,
    padding: "2px 4px",
    alignItems: "center",
    backgroundColor: focused
      ? theme.palette.common.white
      : alpha(theme.palette.common.black, 0.1),
    ...(focused && { boxShadow: theme.shadows[2] }),
  };
});

const SearchEndAdornment = styled("div")({
  // We use a position absolute to support wrapping tags.
  position: "absolute",
  right: 0,
  display: "contents",
});

const SearchItem = styled(MenuItem)(({ theme }) => ({
  borderLeft: "2px solid #fff",
  cursor: "default",
  "&:hover": {
    borderLeftColor: theme.palette.primary.light,
  },
}));

const RemoveItemIcon = styled(ListItemIcon)(() => ({
  cursor: "pointer",
}));

const LightTooltip = styled(({ className, ...props }: TooltipProps) => (
  <Tooltip {...props} classes={{ popper: className }} />
))<StyledPaperProps>(({ theme }) => ({
  [`& .${tooltipClasses.tooltip}`]: {
    backgroundColor: theme.palette.common.white,
    color: "rgba(0, 0, 0, 0.87)",
    boxShadow: theme.shadows[1],
    fontSize: 11,
  },
}));

const SearchPopper = styled(Popper)<StyledPaperProps>(({ theme }) => ({
  zIndex: theme.zIndex.modal,
  backgroundColor: "red",
  border: "1px solid blue",
}));

const SearchPropTypes = {
  placeholderLabel: PropTypes.object,
  clearLabel: PropTypes.object,
  searchLabel: PropTypes.object,
  showSearchOptions: PropTypes.object,
  removeLabel: PropTypes.object,
  open: PropTypes.bool,
  PopperComponent: PropTypes.elementType,
};

export type SearchTypes = InferProps<typeof SearchPropTypes>;

interface SearchTypesWithoutDefaultProps
  extends Omit<
    SearchTypes,
    | "placeholderLabel"
    | "clearLabel"
    | "searchLabel"
    | "showSearchOptions"
    | "removeLabel"
    | "PopperComponent"
  > {}

const Search = (props: SearchTypesWithoutDefaultProps) => {
  const {
    placeholderLabel = messages.placeholder,
    clearLabel = messages.clearLabel,
    searchLabel = messages.searchLabel,
    showSearchOptions = messages.showSearchOptions,
    removeLabel = messages.removeLabel,
    open,
    PopperComponent = Popper,
  } = { ...props };

  const {
    getRootProps,
    getInputProps,
    getListboxProps,
    getOptionProps,
    groupedOptions,
    focused,
    dirty: hasClearIcon,
    getClearProps,
    popupOpen,
    anchorEl,
    setAnchorEl,
  } = useAutocomplete({
    id: "defaultSearch",
    options: top100Films,
    getOptionLabel: (option) => option.title,
  });

  console.log("getRootProps()", getRootProps());

  const intl = useIntl();

  const renderClearIcon = (
    <IconButton
      {...getClearProps()}
      sx={{ p: 0 }}
      aria-label={clearLabel && intl.formatMessage(clearLabel)}
    >
      <CancelRoundedIcon />
    </IconButton>
  );
  const renderDriver = (
    <Divider
      sx={{
        height: 20,
        m: 0.5,
      }}
      orientation="vertical"
    />
  );

  const renderSearch = (
    <InputAdornment position="start">
      <IconButton
        aria-label={intl.formatMessage(searchLabel)}
        onClick={() => console.log("basic search")}
      >
        <SearchIcon onMouseDown={(e) => e.preventDefault()} />
      </IconButton>
    </InputAdornment>
  );
  const renderShowSearchOptions = (
    <IconButton
      onClick={() => console.log("ShowSearchOptions")}
      sx={{ p: "5px" }}
      aria-label={intl.formatMessage(showSearchOptions)}
    >
      <TuneIcon />
    </IconButton>
  );

  return (
    <React.Fragment>
      <SearchRoot
        {...getRootProps()}
        focused={focused}
        elevation={focused ? 1 : 0}
      >
        <InputBase
          fullWidth
          startAdornment={renderSearch}
          endAdornment={
            <SearchEndAdornment>
              {hasClearIcon && renderClearIcon}
              {hasClearIcon && renderDriver}
              {renderShowSearchOptions}
            </SearchEndAdornment>
          }
          ref={setAnchorEl}
          inputProps={{
            "aria-label": intl.formatMessage(placeholderLabel),
            placeholder: intl.formatMessage(placeholderLabel),
            ...getInputProps(),
          }}
        />
      </SearchRoot>
      {popupOpen && anchorEl ? (
        <SearchPopper as={PopperComponent} anchorEl={anchorEl} open>
          SearchPopperaaaaaaaaaa
        </SearchPopper>
      ) : null}
      {/*   {groupedOptions.length > 0 ? (
        <Listbox {...getListboxProps()}>
          {(groupedOptions as typeof top100Films).map((option, index) => (
            <SearchItem {...getOptionProps({ option, index })}>
              <ListItemIcon>
                <HistoryIcon fontSize="small" />
              </ListItemIcon>
              <ListItemText>
                <Typography variant="inherit" noWrap>
                  {option.title}
                </Typography>
              </ListItemText>
              <LightTooltip
                followCursor
                placement="right"
                title={removeLabel ? intl.formatMessage(removeLabel) : ""}
              >
                <RemoveItemIcon>
                  <CloseIcon fontSize="small" />
                </RemoveItemIcon>
              </LightTooltip>
            </SearchItem>
          ))}
        </Listbox>
      ) : null}*/}
      {/*{groupedOptions.length > 0 ? (
        <ul {...getListboxProps()}>
          {(groupedOptions as typeof top100Films).map((option, index) => (
            <li {...getOptionProps({ option, index })}>{option.title}</li>
          ))}
        </ul>
      ) : null}*/}
    </React.Fragment>
  );
};

Search.propTypes = SearchPropTypes;

const messages = defineMessages({
  placeholder: {
    id: "ra.defaultSearch.placeholder",
    defaultMessage: "Search...",
  },
  searchLabel: {
    id: "ra.defaultSearch.search.label",
    defaultMessage: "Search",
  },
  clearLabel: {
    id: "ra.defaultSearch.clear.label",
    defaultMessage: "Clear search",
  },
  showSearchOptions: {
    id: "ra.defaultSearch.show.search.options.label",
    defaultMessage: "Show search options",
  },
  removeLabel: {
    id: "ra.defaultSearch.remove.label",
    defaultMessage: "Remove from history",
  },
});

// Top 100 films as rated by IMDb users. http://www.imdb.com/chart/top
const top100Films = [
  { title: "The Shawshank Redemption", year: 1994 },
  { title: "The Godfather", year: 1972 },
  { title: "The Godfather: Part II", year: 1974 },
  { title: "The Dark Knight", year: 2008 },
  { title: "12 Angry Men", year: 1957 },
  { title: "Schindler's List", year: 1993 },
  { title: "Pulp Fiction", year: 1994 },
];

export default Search;
