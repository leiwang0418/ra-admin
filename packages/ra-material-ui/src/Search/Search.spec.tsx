import React from "react";
import { render, screen, fireEvent } from "../utils/test-utils";
import Search from "./Search";

describe("snapshot test", () => {
	it("render basic Search", () => {
		const { container } = render(<Search />);

		expect(container).toMatchSnapshot();

		expect(
			screen.getByLabelText(/Search.../i, { selector: "input" })
		).toBeInTheDocument();

		expect(
			screen.getByRole("button", { name: "Search" })
		).toBeInTheDocument();

		expect(
			screen.getByRole("button", { name: "Show search options" })
		).toBeInTheDocument();

		expect(
			screen.queryByRole("button", { name: "Clear search" })
		).not.toBeInTheDocument();
	});

	it("render has clear button Search", () => {
		const { container } = render(<Search />);
		const search = screen.getByLabelText(/Search.../i, {
			selector: "input",
		});

		expect(
			screen.queryByRole("button", { name: "Clear search" })
		).not.toBeInTheDocument();

		fireEvent.change(search, { target: { value: "lei" } });

		expect(screen.getByDisplayValue("lei") === search).toBe(true);
		expect(
			screen.queryByRole("button", { name: "Clear search" })
		).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});

// todo Act not test
// describe('act test', () => {
// 	it('should ', () => {
		
// 	})
// });
