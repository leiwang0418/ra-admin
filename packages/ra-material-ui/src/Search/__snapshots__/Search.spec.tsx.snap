// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test render basic Search 1`] = `
<div>
  <div>
    <div
      aria-expanded="false"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-124oeyu-MuiPaper-root"
      role="combobox"
    >
      <div
        class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-adornedStart MuiInputBase-adornedEnd css-luel6v-MuiInputBase-root"
      >
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionStart css-ittuaa-MuiInputAdornment-root"
        >
          <span
            class="notranslate"
          >
            â€‹
          </span>
          <button
            aria-label="Search"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <input
          aria-autocomplete="list"
          aria-label="Search..."
          autocapitalize="none"
          autocomplete="off"
          class="MuiInputBase-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-yz9k0d-MuiInputBase-input"
          id="use-autocomplete-demo"
          placeholder="Search..."
          spellcheck="false"
          type="text"
          value=""
        />
        <div
          class="css-i45yfx"
        >
          <button
            aria-label="Show search options"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-14cutrz-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="TuneIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17v2h6v-2H3zM3 5v2h10V5H3zm10 16v-2h8v-2h-8v-2h-2v6h2zM7 9v2H3v2h4v2h2V9H7zm14 4v-2H11v2h10zm-6-4h2V7h4V5h-4V3h-2v6z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
